src/signer/wallet.rs

```diff
@@ -0,0 +1,117 @@
+use universal_wallet::{
+    prelude::*,
+    contents::Content,
+};
+use super::KeyManager;
+use crate::{
+    keys::PublicKey,
+    error::Error,
+};
+
+pub const CURRENT: &str = "current";
+const NEXT: &str = "next";
+pub const CRYPTO: &str = "crypto";
+
+/// Helper function to populate wallet with properly tagged
+/// signing and crypto key pairs
+///
+pub fn incept_keys(wallet: &mut UnlockedWallet) -> Result<(), Error> {
+    let mut current = KeyPair::random_pair(KeyType::Ed25519VerificationKey2018)?;
+    current.public_key.controller = vec!(CURRENT.into());
+    wallet.set_content(CURRENT, Content::KeyPair(current));
+    let mut next = KeyPair::random_pair(KeyType::Ed25519VerificationKey2018)?;
+    next.public_key.controller = vec!(NEXT.into());
+    wallet.set_content(NEXT, Content::KeyPair(next));
+    let mut crypto = KeyPair::random_pair(KeyType::X25519KeyAgreementKey2019)?;
+    crypto.public_key.controller = vec!(CRYPTO.into());
+    wallet.set_content(CRYPTO, Content::KeyPair(crypto));
+    Ok(())
+}
+
+impl KeyManager for UnlockedWallet {
+    fn sign(&self, msg: &[u8]) -> Result<Vec<u8>, Error> {
+        // FIXME: figure out how to fetch signing key with limited trait
+        Ok(UnlockedWallet::sign_raw(&self, CURRENT, msg)?)
+    }
+
+    // TODO: do we really need this trait not to fail or option out?
```

Comment:
It would be nice to return Result or Option here.

Reply:
that's how `KeyManager` is implemented. if that call is not an option or result - it's unfailable, right? otherwise - that's the trait method sig which needs to be changed :)